drop-transfer/src/ws/events.rs

```diff
@@ -404,14 +438,112 @@ impl FileEventTx<OutgoingTransfer> {
     }
 }
 
+impl<T: Transfer> TransferEventTx<T> {
+    async fn emit_ongoing(&self, event: Event) {
+        let lock = self.inner.lock().await;
+
+        if let TransferState::Terminated = lock.state {
+            return;
+        }
+
+        lock.tx.emit(event);
+    }
+
+    async fn stop_with_moose(&self, event: Event, status: Result<(), i32>) {
+        let mut lock = self.inner.lock().await;
+
+        if let TransferState::Terminated =
+            std::mem::replace(&mut lock.state, TransferState::Terminated)
+        {
+            return;
+        }
+
+        lock.moose.event_transfer_state(TransferStateEventData {
+            protocol_version: 0,
+            transfer_id: self.xfer.id().to_string(),
+            result: status.map_or_else(|err| err, |()| MOOSE_STATUS_SUCCESS),
+        });
+
+        lock.tx.emit(event);
+    }
```

Comment:
I think this function could be simplified and `moose` could be extracted to the call site. This way it would be more uniform with other functions. And maybe even `stop()` `stop_with_moose()` `emit_ongoing()` could be squashed into the same function as their body would be the same. What do you think?

Reply:
Sure, moose can be extracted. But `emit_ongoing()` and `stop()` cannot be squashed because the first one only checks if it's terminated and the latter set the state to `Termianted`

Reply:
Done